### PR TITLE
Add comment_visibility parameter for comment operation for jira module

### PR DESCRIPTION
**Pull request by momcilo78**
##### SUMMARY
This change  includes the additional parameter for the jira module. When operation is set to comment, comment_visibility will be used to limit the visibility of the comment. JIRA itself provide 2 means for declaring the visibility: group and role. The parameter needs to include the type and value fields. For this reason the parameter is not simple but rather a dict.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
jira

##### ADDITIONAL INFORMATION
For more information on visibility of comments in JIRA please see Visibility object under following section:
https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-comments/#api-rest-api-2-issue-issueidorkey-comment-post

### CODE REVIEW

**Comment by momcilo78** [comment]
I just realized that my fork was not up-to-date...
It appears the modules got switched from functional to object oriented, please hold this on...

File: plugins/modules/web_infrastructure/jira.py [hunk 1]
```diff
@@ -82,6 +82,12 @@
     description:
      - The comment text to add.
 
+  comment_visibility:
+    type: dict
+    required: false
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jira.py:87]:
```suggestion
    version_added: 3.2.0
```

File: plugins/modules/web_infrastructure/jira.py [hunk 2]
```diff
@@ -82,6 +82,12 @@
...
+    description:
+      - used to specify comment comment visibility. Argument is dictionary containing type and value. Type can be [group, role]. Please see: https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-comments/#api-rest-api-2-issue-issueidorkey-comment-post
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jira.py:89]:
```suggestion
      - Used to specify comment comment visibility.
      - See U(https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-comments/#api-rest-api-2-issue-issueidorkey-comment-post) for details.
    options:
      type:
        description:
          - ...
        type: str
        required: true
        choices: [group, role]
      value:
        description:
          - ...
        type: str
        required: true
```

File: plugins/modules/web_infrastructure/jira.py [hunk 3]
```diff
@@ -388,6 +406,10 @@ def comment(restbase, user, passwd, params):
         'body': params['comment']
     }
 
+    # if comment_visibility is specified restrict visibility
+    if 'comment_visibility' in params.keys():
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jira.py:410]: That condition is *always* true. You should rather do:
```suggestion
    if params['comment_visibility'] is not None:
```

File: plugins/modules/web_infrastructure/jira.py [hunk 4]
```diff
@@ -501,6 +523,7 @@ def main():
             issuetype=dict(type='str', ),
             issue=dict(type='str', aliases=['ticket']),
             comment=dict(type='str', ),
+            comment_visibility=dict(type='dict', ),
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/jira.py:526]:
```suggestion
            comment_visibility=dict(type='dict', suboptions=dict(
                type=dict(type='str', choices=['group', 'role'], required=True),
                value=dict(type='str', required=True),
            )),
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! You need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to). I added some comments for better validation below.

CC @russoz

**Comment by felixfontein** [comment]
Please note that you need to rebase your PR, since your PR is based on an outdated version of the module.

**Comment by felixfontein** [comment]
> I just realized that my fork was not up-to-date...
> It appears the modules got switched from functional to object oriented, please hold this on...

Also just noticed that ;)

**Comment by momcilo78** [comment]
> 
> 
> > I just realized that my fork was not up-to-date...
> > It appears the modules got switched from functional to object oriented, please hold this on...
> 
> Also just noticed that ;)

It looks like I got stuck deep inside git for rebase, but I ran out of time today.

**Pull request closed without merging**